llvm/lib/Target/RISCV/RISCVInstrInfo.cpp

```diff
@@ -1647,6 +1870,9 @@ bool RISCVInstrInfo::hasReassociableSibling(const MachineInstr &Inst,
 
 bool RISCVInstrInfo::isAssociativeAndCommutative(const MachineInstr &Inst,
                                                  bool Invert) const {
+  if (isVectorAssociativeAndCommutative(Inst, Invert))
+    return true;
+
```

Comment:
Nit, you could also use `RISCV::getRVVMCOpcode` and then just add `RISCV::VADD_VV` etc. to the existing switch below, something like

```suggestion
  unsigned Opc = Inst.getOpcode();
  if (unsigned RVVOpc = RISCV::getRVVMCOpcode(Opc))
    Opc = RVVOpc;
```

Then I don't think we would need the separate vector function